src/modules/python/pbs_hooks/PBS_power.PY

```diff
@@ -392,23 +450,48 @@ if this_event.type == pbs.PERIODIC:
                 power.ramp_down(poweroff_vnlist)
             if poweron_vnlist:
                 power.ramp_up(poweron_vnlist)
-            up_state = pbs.ND_FREE
         else:
             # Power on/off nodes
             if poweroff_vnlist:
                 power.power_off(poweroff_vnlist)
             if poweron_vnlist:
                 power.power_on(poweron_vnlist)
-            up_state = pbs.ND_STALE
+            ready_nodes = set()
+            if sleep_node_list:
+                ready_nodes = power.power_status(sleep_node_list)
+                # For the nodes which are up remove ND_SLEEP
+                for host in ready_nodes:
+                    for vn in nodes[host]["vnodes"]:
+                        prev_state = vn.state
+                        vn.state = prev_state & ~(pbs.ND_SLEEP)
+                        vn.last_used_time = time_now
+            # nodes which are still booting, write them to
+            # node file for status check in next iteration.
+            if sleep_node_list or poweron_vnlist:
+                try:
+                    fd = open(node_file, 'w')
+                    sleeping_nodes = poweron_vnlist
+                    # Look for still booting nodes from previous iteration.
+                    for n in sleep_node_list:
+                        if n not in ready_nodes:
+                            sleeping_nodes.append(n)
+                    data = ','.join(sleeping_nodes)
+                    fd.write(data)
+                    fd.close()
+                except IOError:
+                    raise Exception("I/O error reading node file")
+                except:
+                    raise Exception("Error reading node file")
```

Comment:
Don't use a blanket except.  Narrow down the focus.

Reply:
Done.